extensions/ql-vscode/src/view/alert-table.tsx

```diff
@@ -72,7 +74,8 @@ export function getPathRelativeToSourceLocationPrefix(sourceLocationPrefix: stri
 export class PathTable extends React.Component<PathTableProps, PathTableState> {
   constructor(props: PathTableProps) {
     super(props);
-    this.state = { expanded: {} };
+    this.state = { expanded: {}, selectedPathNode: undefined };
+    this.handleNavigationEvent = this.handleNavigationEvent.bind(this);
```

Comment:
I almost prefer to have the `.bind(this)`s at the callsite, or else give the property the `.bind()`ed method a different name from the original method name. Do you have a strong opinion here? I haven't seen this pattern before but am willing to defer if you feel like it's idiomatic.

Reply:
I'd say it's quite common. It's recommended [on reactjs.org](https://reactjs.org/docs/handling-events.html), with the other alternative being to make the method into a field instead.

> We generally recommend binding in the constructor or using the class fields syntax, to avoid this sort of performance problem.